src/sentry/api/endpoints/project_performance_issue_settings.py

```diff
@@ -0,0 +1,75 @@
+from rest_framework import serializers, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features, projectoptions
+from sentry.api.bases.project import ProjectEndpoint, ProjectSettingPermission
+
+MAX_VALUE = 2147483647
+SETTINGS_PROJECT_OPTION_KEY = "sentry:performance_issue_settings"
+
+
+class ProjectPerformanceIssueSettingsSerializer(serializers.Serializer):
+    n_plus_one_db_detection_rate = serializers.FloatField(required=False, min_value=0, max_value=1)
+    n_plus_one_db_issue_rate = serializers.FloatField(required=False, min_value=0, max_value=1)
+    n_plus_one_db_count = serializers.IntegerField(required=False, min_value=0, max_value=MAX_VALUE)
+    n_plus_one_db_duration_threshold = serializers.IntegerField(
+        required=False, min_value=0, max_value=MAX_VALUE
+    )
+
+
+class ProjectPerformanceIssueSettingsEndpoint(ProjectEndpoint):
+    private = True  # TODO: Remove after EA.
+    permission_classes = (ProjectSettingPermission,)
+
+    def has_feature(self, project, request) -> bool:
+        return features.has(
+            "organizations:performance-view", project.organization, actor=request.user
+        ) and features.has(
+            "organizations:performance-issues", project.organization, actor=request.user
+        )
+
+    def get(self, request: Request, project) -> Response:
+        """
+        Retrieve performance issue settings
+        ``````````````````
+
+        Return settings for performance issues
+
+        :pparam string organization_slug: the slug of the organization the
+                                          project belongs to.
+        :pparam string project_slug: the slug of the project to configure.
+        :auth: required
+        """
+
+        if not self.has_feature(project, request):
+            return self.respond(status=status.HTTP_404_NOT_FOUND)
```

Comment:
Shouldn't we use an unauthorized status code in this case instead?

Reply:
Yeah I'll fix that in a follow up